tests/integration_test/core_mqtt_system_test.c

```diff
@@ -1688,22 +1698,18 @@ TEST( coreMQTT_Integration, Publish_With_Retain_Flag )
                                                     MQTT_GetPacketId( &context ) ) );
     /* Complete the QoS 1 PUBLISH operation. */
     TEST_ASSERT_FALSE( receivedPubAck );
-    TEST_ASSERT_EQUAL( MQTTSuccess,
-                       MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
-    TEST_ASSERT_TRUE( receivedPubAck );
+    waitForPacket( &receivedPubAck, MQTTSuccess );
 
     /* Subscribe to the same topic that we published the message to.
      * The broker should send the "retained" message with the "retain" flag set. */
     TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic(
                            &context, TEST_MQTT_TOPIC, MQTTQoS1 ) );
     TEST_ASSERT_FALSE( receivedSubAck );
-    TEST_ASSERT_EQUAL( MQTTSuccess,
-                       MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
-    TEST_ASSERT_TRUE( receivedSubAck );
+    waitForPacket( &receivedSubAck, MQTTSuccess );
 
     /* Make sure that the library invoked the event callback with the incoming PUBLISH from
      * the broker containing the "retained" flag set. */
-    TEST_ASSERT_TRUE( receivedRetainedMessage );
+    waitForPacket( &receivedRetainedMessage, MQTTSuccess );
```

Comment:
Is this the only instance where the flag we're waiting on isn't guaranteed to be `false` before calling the function? Otherwise, I see that `TEST_ASSERT_FALSE` is called before every other invocation of `waitForPacket`, so could be moved into it.

This isn't as pretty, but it would save one line for every other call of `waitForPacket`:
```c
/* If the retained message was not received in the last wait, wait again. */
if( !receivedRetainedMessage )
{
    waitForPacket( &receivedRetainedMessge, MQTTSuccess );
}
```

Reply:
sure thanks

Reply:
No that makes sense. I think it looks fine.